camayoc/tests/qcs/test_profiles.py

```diff
@@ -149,26 +244,59 @@ def test_read_all(cleanup):
            created network profiles
         3) Confirm that all profiles are in the list.
     :expectedresults: All profiles are present in data returned by API.
-    :caseautomation: notautomated
+    :caseautomation: automated
     """
-    pass
+    created_profs = []
+    for i in range(random.randint(2, 20)):
+        # gen_valid_profs will take care of cleanup
+        created_profs.append(
+            gen_valid_profile(
+                cleanup,
+                shared_client,
+                'localhost'))
+    server_profs = NetworkProfile().list().json()
+    for sp in server_profs:
+        found = False
+        for i, cp in enumerate(created_profs):
+            if sp['id'] == cp.fields()['id']:
+                if cp.equivalent(sp):
+                    found = True
```

Comment:
If you have found the network profile then you can drop it from created_profs here. Also you don't need the index (i assigned by enumerate) you can call remove on the networkprofile object: `created_profs.remove(cp)`.